File: CodeSchool.Domain/UserLessonProgress.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace CodeSchool.Domain
{
    public class UserLessonProgress
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public int LessonId { get; set; }
        public Guid UserChapterProgressId { get; set; }
        public Guid UserId { get; set; }
        public bool IsPassed { get; set; }
        public string Code { get; set; }
        public DateTime UpdatedDt { get; set; }
    }
}

File: CodeSchool.Web/ClientApp/app/models/userlessonprogress.ts
import { Mapper } from "../utils/helpers";

export class UserLessonProgressModel {
    constructor(model?: UserLessonProgressModel) {
        if (model) {
            Mapper.map(model, this);
        }
    }

    id: string;
    lessonId: number;
    userChapterProgressId: string;
    userId: string;
    isPassed: boolean;
    code: string;
    updatedDt: any;
}